lib/reek/spec/smell_matcher.rb

```diff
@@ -0,0 +1,57 @@
+module Reek
+  module Spec
+    #
+    # Matches a +SmellWarning+ object agains a smell type and hash of attributes.
+    #
+    class SmellMatcher
+      attr_reader :smell_warning
+
+      COMPARABLE_ATTRIBUTES = %i(message lines context source)
+
+      def initialize(smell_warning)
+        @smell_warning = smell_warning
+      end
+
+      def matches?(klass, attributes = {})
+        matches_smell_type?(klass) && matches_attributes?(attributes)
+      end
+
+      def matches_smell_type?(klass)
+        smell_classes.include?(klass.to_s)
+      end
+
+      def matches_attributes?(attributes)
+        check_attributes_comparability(attributes)
+
+        main_attributes = attributes.slice(*COMPARABLE_ATTRIBUTES)
+        parameter_attributes = attributes.except(*COMPARABLE_ATTRIBUTES)
```

Comment:
'cause it spanks too many monkeys!

But seriously, having it in our tests proves a bit dangerous because we might accidentally use it in our library code.